src/api.py

```diff
@@ -67,7 +73,13 @@ def _search_posts_query(username: str, search_content: str) -> Optional[dict]:
                 },
             ],
         }
+    else:  # avoid an empty $or clause which will cause an error
```

Comment:
Bit of a nit, but I would prefer to have this as a guard clause rather than an if/else, e.g.:
```python
if not username:
    return None
# other stuff is no longer inside an if/else block
```